src/tree/scaling/ScaleItem.ts

```diff
@@ -43,22 +47,48 @@ export class ScaleItem implements ContainerAppsItem {
     getTreeItem(): TreeItem {
         return {
             id: this.id,
-            label: localize('scaling', 'Scaling'),
+            label: this.hasUnsavedChanges() ? `${scaling}*` : scaling,
             contextValue: ScaleItem.contextValue,
             iconPath: treeUtils.getIconPath('scaling'),
             collapsibleState: TreeItemCollapsibleState.Collapsed,
         }
     }
 
-    async getChildren?(): Promise<TreeElementBase[]> {
+    getChildren(): TreeElementBase[] {
+        let scale: Scale | undefined;
+
+        if (this.hasUnsavedChanges()) {
+            scale = ext.revisionDraftFileSystem.parseRevisionDraft(this)?.scale;
+        } else if (this.containerApp.revisionsMode === KnownActiveRevisionsMode.Single) {
+            scale = this.containerApp.template?.scale;
+        } else {
+            scale = this.revision.template?.scale;
+        }
+
         return [
             createGenericElement({
                 label: localize('minMax', 'Min / max replicas'),
-                description: `${this.scale?.minReplicas ?? 0} / ${this.scale?.maxReplicas ?? 0}`,
+                description: `${scale?.minReplicas ?? 0} / ${scale?.maxReplicas ?? 0}`,
                 contextValue: minMaxReplicaItemContextValue,
                 iconPath: new ThemeIcon('dash'),
             }),
-            createScaleRuleGroupItem(this.subscription, this.containerApp, this.revision),
-        ]
+            createScaleRuleGroupItem(this.subscription, this.containerApp, this.revision, scale?.rules ?? []),
+        ];
+    }
+
+    private hasUnsavedChanges(): boolean {
+        const scaleDraftTemplate = ext.revisionDraftFileSystem.parseRevisionDraft(this)?.scale;
+        if (!scaleDraftTemplate) {
+            return false;
+        }
+
+        if (this.containerApp.revisionsMode === KnownActiveRevisionsMode.Single) {
+            return !!this.containerApp.template?.scale && !isDeepEqual(this.containerApp.template.scale, scaleDraftTemplate);
+        } else {
+            // We only care about showing changes to descendants of the revision draft item when in multiple revisions mode
```

Comment:
This code segment will be introduced in the multiple revisions PR, commented out for now with a return placeholder